spec/intl.html

```diff
@@ -292,46 +292,62 @@ <h1>PartitionDateTimePattern ( _dateTimeFormat_, _x_ )</h1>
         1. <ins>If IsTemporalObject(_x_) is *true*, then</ins>
           1. <mark>TODO: _era_, _dayPeriod_, _fractionalSecondDigits_.</mark>
           1. <ins>If _x_ has an [[InitializedTemporalDate]] internal slot, then</ins>
-            1. <ins>Let _calendar_ be ? CalendarToString(_x_.[[Calendar]]).</ins>
-            1. <ins>If _calendar_ is not *"iso8601"* and not equal to _dateTimeFormat_.[[Calendar]], throw a *RangeError* exception.</ins>
             1. <ins>Let _pattern_ be _dateTimeFormat_.[[TemporalPlainDatePattern]].</ins>
-            1. <ins>Let _tm_ be { [[weekday]]: ! ToDayOfWeek(_x_.[[ISOYear]], _x_.[[ISOMonth]], _x_.[[ISODay]]), [[year]]: _x_.[[ISOYear]], [[month]]: _x_.[[ISOMonth]], [[day]]: _x_.[[ISODay]] }.</ins>
-          1. <ins>If _x_ has an [[InitializedTemporalYearMonth]] internal slot, then</ins>
             1. <ins>Let _calendar_ be ? CalendarToString(_x_.[[Calendar]]).</ins>
-            1. <ins>If _calendar_ is not equal to _dateTimeFormat_.[[Calendar]], throw a *RangeError* exception.</ins>
+            1. <ins>If _calendar_ is _dateTimeFormat_.[[Calendar]], then</ins>
+              1. <ins>Let _calendarOverride_ be _x_.[[Calendar]].</ins>
+            1. <ins>Else if _calendar_ is *"iso8601"*, then</ins>
+              1. <ins>Let _calendarOverride_ be ? CalendarFrom(_dateTimeFormat_.[[Calendar]]).</ins>
+              1. <mark>Note that this calls `Temporal.Calendar.from`, which may be author code. Desirable?.</mark>
```

Comment:
Not entirely sure what's correct, but I think you could use GetBuiltinCalendar here? _dateTimeFormat_.[[Calendar]] is always a built-in calendar, and I think we don't want user code to be able to change the behaviour of toLocaleString by monkeypatching Temporal.Calendar.from().